modules/videoio/src/cap_v4l.cpp

```diff
@@ -466,12 +466,16 @@ static int autosetup_capture_mode_v4l2(CvCaptureCAM_V4L* capture) {
             V4L2_PIX_FMT_SN9C10X,
             V4L2_PIX_FMT_SBGGR8,
             V4L2_PIX_FMT_SGBRG8,
-            V4L2_PIX_FMT_RGB24
+            V4L2_PIX_FMT_RGB24,
+            V4L2_PIX_FMT_Y16
     };
 
     for (size_t i = 0; i < sizeof(try_order) / sizeof(__u32); i++) {
         capture->palette = try_order[i];
         if (try_palette_v4l2(capture)) {
+            if (capture->palette == V4L2_PIX_FMT_Y16) {
+                capture->convert_rgb = false;
+            }
```

Comment:
Problem is here.
`capture->convert_rgb = false` should be changed explicitly by user only (via set property call with CV_CAP_PROP_CONVERT_RGB = false).